File: lib_network/src/main/java/com/android/sdk/net/CommonBuilder.java
package com.android.sdk.net;

import androidx.annotation.MainThread;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.android.sdk.net.core.provider.ErrorBodyParser;
import com.android.sdk.net.core.provider.ErrorMessage;
import com.android.sdk.net.core.provider.PlatformInteractor;
import com.android.sdk.net.coroutines.CoroutinesResultPostProcessor;
import com.android.sdk.net.rxjava2.RxResultPostTransformer;

public class CommonBuilder {

    private final NetContext mNetContext;

    CommonBuilder(NetContext netContext) {
        mNetContext = netContext;
    }

    private final CommonProviderImpl mCommonProvider = new CommonProviderImpl();

    public CommonBuilder errorMessage(@NonNull ErrorMessage errorMessage) {
        mCommonProvider.mErrorMessage = errorMessage;
        return this;
    }

    public CommonBuilder platformInteractor(@NonNull PlatformInteractor platformInteractor) {
        mCommonProvider.mPlatformInteractor = platformInteractor;
        return this;
    }

    @MainThread
    public NetContext setUp() {
        mCommonProvider.checkRequirement();
        mNetContext.initCommonProvider(mCommonProvider);
        return mNetContext;
    }

}

File: lib_network/src/main/java/com/android/sdk/net/CommonBuilder.kt
package com.android.sdk.net

import androidx.annotation.MainThread
import com.android.sdk.net.core.provider.ErrorMessageConverter
import com.android.sdk.net.core.provider.PlatformInteractor

class CommonBuilder internal constructor(private val netContext: NetContext) {

    private val commonProvider = CommonProviderImpl()

    fun errorMessageConverter(errorMessageConverter: ErrorMessageConverter): CommonBuilder {
        commonProvider.mErrorMessageConverter = errorMessageConverter
        return this
    }

    fun platformInteractor(platformInteractor: PlatformInteractor): CommonBuilder {
        commonProvider.mPlatformInteractor = platformInteractor
        return this
    }

    @MainThread
    internal fun setup(): NetContext {
        commonProvider.checkRequirement()
        netContext.initCommonProvider(commonProvider)
        return netContext
    }

}